src/AppWrapper.js

```diff
@@ -0,0 +1,88 @@
+import { apiFetchOrganisationUnitLevels } from '@dhis2/analytics'
+import { useConfig, useDataEngine } from '@dhis2/app-runtime'
+import { D2Shim } from '@dhis2/app-runtime-adapter-d2'
+import React, { useState, useEffect, useCallback } from 'react'
+import { Provider as ReduxProvider } from 'react-redux'
+import thunk from 'redux-thunk'
+import App from './components/App'
+import UserSettingsProvider, {
+    UserSettingsCtx,
+} from './components/UserSettingsProvider'
+import configureStore from './configureStore'
+import metadataMiddleware from './middleware/metadata'
+import history from './modules/history'
+import './locales'
+
+const AppWrapper = () => {
+    const { baseUrl } = useConfig()
+    const engine = useDataEngine()
+    const store = configureStore([
+        thunk.withExtraArgument(engine),
```

Comment:
We might be able to avoid this if we move away from having fetch functions in the `api` folder and use the hooks directly in the components instead.